src/library/scala/Option.scala

```diff
@@ -270,7 +270,7 @@ sealed abstract class Option[+A] extends Product with Serializable {
    *  // Returns None because the partial function doesn't cover the case.
    *  Some("ftp") collect {case "http" => "HTTP"}
    *
-   *  // Returns None because None is passed to the collect method.
+   *  // Returns None because there is no value to pass to the partial function.
```

Comment:
@dickwall I like this comment but I think the optimal location for it would be as part of a replacement for the existing non-example comments.

I've tried to preserve the drill-down nature of the existing example comments by taking a slight variation on @som-snytt's suggestion which follows this template,

    Return XX because <implementation detail>

Admittedly comments that restate the implementation are not ideal but the scope of this commit is an error fix rather than a wider improvement.

WDYT?